harness/determined/common/schemas/expconf/_validate.py

```diff
@@ -14,7 +14,9 @@ def make_validator(url: Optional[str] = None, complete: Optional[bool] = False)
         url = "http://determined.ai/schemas/expconf/v1/experiment.json"
 
     global _validators
-    if url in _validators:
+    # Need a new validator to be made for completeness validation. If we
+    # used the cached sanity validator, we wouldn't check `eventuallyRequire`, etc.
+    if url in _validators and not complete:
```

Comment:
@rb-determined-ai instead of removing caching altogether for `completeness` validators, should we perhaps cache it as `url+"complete"`?

Reply:
just use two separate caches, `_sanity_validators` and `_completeness_validators`, or let `_validators` be a map of bools to maps of urls to validators, like `_validators[optional][url]` or something like that.